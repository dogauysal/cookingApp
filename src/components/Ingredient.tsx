import { Box, Flex, ListItem, Text } from "@react-native-material/core";
import React from "react";
import { StyleSheet } from "react-native";

interface IProps {
    ingredient: string
}

const Ingredient: React.FC<IProps> = ({
    ingredient
}) => {
    return (
        <Flex style={{ flexDirection: "row", alignContent: "flex-start" }}>
            <Text>- {ingredient}</Text>
        </Flex>
    )
}

const styles = StyleSheet.create({
    itemText: {

    }
})

export default Ingredient