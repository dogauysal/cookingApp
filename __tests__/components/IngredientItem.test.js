import React from "react";
import renderer from 'react-test-renderer';
import Ingredient from "../../src/components/Ingredient";



describe('ingredient item', () => {
    test('ingredientitem renders correctly', () => {
        const ingredient = "cheese"

        const ingredientItem = renderer.create(<Ingredient ingredient={ingredient} />).toJSON();
        expect(ingredientItem).toMatchSnapshot();
    })
})

