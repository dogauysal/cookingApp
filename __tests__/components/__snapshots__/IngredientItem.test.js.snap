// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ingredient item ingredientitem renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderBottomColor": undefined,
        "borderBottomEndRadius": undefined,
        "borderBottomLeftRadius": undefined,
        "borderBottomRightRadius": undefined,
        "borderBottomStartRadius": undefined,
        "borderBottomWidth": undefined,
        "borderColor": undefined,
        "borderEndColor": undefined,
        "borderEndWidth": undefined,
        "borderLeftColor": undefined,
        "borderLeftWidth": undefined,
        "borderRadius": undefined,
        "borderRightColor": undefined,
        "borderRightWidth": undefined,
        "borderStartColor": undefined,
        "borderStartWidth": undefined,
        "borderStyle": undefined,
        "borderTopColor": undefined,
        "borderTopEndRadius": undefined,
        "borderTopLeftRadius": undefined,
        "borderTopRightRadius": undefined,
        "borderTopStartRadius": undefined,
        "borderTopWidth": undefined,
        "borderWidth": undefined,
        "bottom": undefined,
        "end": undefined,
        "height": undefined,
        "left": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginEnd": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginStart": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "opacity": undefined,
        "overflow": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingEnd": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingStart": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "start": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      },
      Array [
        Object {
          "alignContent": undefined,
          "alignItems": undefined,
          "alignSelf": undefined,
          "flex": undefined,
          "flexBasis": undefined,
          "flexDirection": undefined,
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": undefined,
          "justifyContent": undefined,
        },
        Object {
          "alignContent": "flex-start",
          "flexDirection": "row",
        },
      ],
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "letterSpacing": 0.5,
        },
        Object {
          "color": "#000000",
        },
        undefined,
      ]
    }
  >
    - 
    cheese
  </Text>
</View>
`;
